halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/AwsCredentialsProfileFactoryBuilder.java

```diff
@@ -40,12 +46,44 @@
 
   @Setter private String profileName = "default";
 
-  @Setter private String accessKeyId;
+  @Setter(AccessLevel.NONE)
+  private String accessKeyId;
 
-  @Setter private String secretAccessKey;
+  @Setter(AccessLevel.NONE)
+  private String secretAccessKey;
 
   @Setter private SpinnakerArtifact artifact;
 
+  public AwsCredentialsProfileFactoryBuilder setAccessKeyId(String accessKeyId) {
+    if (StringUtils.isEmpty(accessKeyId)) {
+      try {
+        this.accessKeyId =
+            DefaultAWSCredentialsProviderChain.getInstance().getCredentials().getAWSAccessKeyId();
```

Comment:
This assumes that the creds are not temporary, for example not AssumeRole creds which would have a session token.  If I am running hal on an EC2 instance, it can pick up creds from the instance role.